pagefaultgames/pokerogue
# [P1] Fixing continue menu option when game finishes

## What are the changes the user will see?
- Continue button will no longer show up if the last and only session game completes

## Why am I making these changes?
- Continue button shows up erroneously if last and only session game completes. If user selects it game freezes/crashes

## What are the changes from a developer perspective?
- UserInfo was not being properly updated in the post end game phase. There were 2 issues:
    1. it would never update for localStorage only runs
    2. for server runs it would try to update UserInfo before the session was cleared

## How to test the changes?
- Start a session making sure it is the only session & make sure it saves. Restart page and continue the run ending it. Once ended the continue button will no longer show up.

## Bug report
- https://github.com/pagefaultgames/pokerogue/issues/4288

## Checklist
- [x] Have I considered writing automated tests for the issue? took a quick look. doesn't look like project is setup to mock and test logic for phases touching updating user data
